group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1949,6 +1949,15 @@ private CoordinatorResult<StreamsGroupHeartbeatResult, CoordinatorRecord> stream
                 reconfigureTopology = true;
             }
 
+            // Check if assignment configurations have changed
```

Comment:
I think this should be moved outside of this if. We can check it right before actually bumping the group epoch, so in a sense, it would become step 3c after step 3b.